quickwit/quickwit-cluster/src/change.rs

```diff
@@ -19,24 +19,56 @@
 
 use std::collections::btree_map::Entry;
 use std::collections::BTreeMap;
+use std::pin::Pin;
+use std::task::{Context, Poll};
 
 use chitchat::{ChitchatId, NodeState};
+use futures::Stream;
+use pin_project::pin_project;
 use quickwit_common::sorted_iter::{KeyDiff, SortedByKeyIterator};
 use quickwit_common::tower::{make_channel, warmup_channel};
 use quickwit_proto::types::NodeId;
+use tokio::sync::mpsc;
+use tokio_stream::wrappers::UnboundedReceiverStream;
 use tonic::transport::Channel;
 use tracing::{info, warn};
 
 use crate::member::NodeStateExt;
 use crate::ClusterNode;
 
+/// Describes a change in the cluster.
 #[derive(Debug, Clone)]
 pub enum ClusterChange {
     Add(ClusterNode),
     Update(ClusterNode),
     Remove(ClusterNode),
 }
 
+#[pin_project]
+pub struct ClusterChangeStream(#[pin] UnboundedReceiverStream<ClusterChange>);
+
+impl ClusterChangeStream {
+    pub fn new_unbounded() -> (Self, mpsc::UnboundedSender<ClusterChange>) {
+        let (change_stream_tx, change_stream_rx) = mpsc::unbounded_channel();
+        (
+            Self(UnboundedReceiverStream::new(change_stream_rx)),
+            change_stream_tx,
+        )
+    }
+}
+
+impl Stream for ClusterChangeStream {
+    type Item = ClusterChange;
+
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        self.project().0.poll_next(cx)
+    }
+}
+
+pub trait ClusterChangeStreamFactory: Clone + Send + 'static {
```

Comment:
Dealing with a `Cluster` object for testing is just too painful when all we want is to send some `ClusterChange` events in a channel. So here you have it.